### ISSUE TITLE
 telescope-fzf-writer.nvim fails after the merge of #709

### ISSUE THREAD

**Issue by mherna** [nvim-telescope/telescope.nvim#748]
### Description
I don't know whether to open this in here or in `nvim-telescope/telescope-fzf-writer.nvim` but after #709 staged_grep() fails to work. I tried seeing if reverting to commit e5fbe6f made staged_grep() work again, and it did. 

**Expected Behavior**
returns search results

**Actual Behavior**
```
 8 || ...ck/packer/start/telescope.nvim/lua/telescope/pickers.lua:953: attempt to index field 'sorter' (a boolean value)↲                            
 9 || Error executing luv callback:↲                                                                                                                 
10 || cannot resume dead coroutine↲
```

### Details

run `require('telescope').extensions.fzf_writer.staged_grep()` and type something.

**Comment by yujinyuz** [comment]
same. can reproduce the issue with `require('telescope').extensions.fzf_writer.files{}`

**Comment by tami5** [comment]
@Conni2461 What a coincidence!! :laughing: 

I guess won't fix :rofl: because fzf-writer will be deprecated.

**Comment by yujinyuz** [comment]
> @Conni2461 What a coincidence!! 😆
> 
> I guess won't fix 🤣 because fzf-writer will be deprecated.

Why won't it get fixed? Is it going to be merged to telescope?

**Comment by Conni2461** [comment]
@tami5 -.- you legit just jinxed it.

I don't know whats wrong with writer sorry (i have never used it). And we will not merge it into telescope because it requires a external dependency.
But i am working on a `telescope-fzf-native.nvim` and i am pretty far and @tami5 just spilled the beans :angry: 

Here are some more spoilers, basically showing how many files match.
![image](https://user-images.githubusercontent.com/15233006/114301296-8294f600-9ac4-11eb-9ae3-0de37cf63a8d.png)
![image](https://user-images.githubusercontent.com/15233006/114301306-8aed3100-9ac4-11eb-92a3-360b54a98eaf.png)
![image](https://user-images.githubusercontent.com/15233006/114301334-96d8f300-9ac4-11eb-86b5-e0321ac657b0.png)

I am still working on glue code which will determine, which alg will be called. I already ported all algorithms:
- fuzzy_match_v2
- fuzzy_match_v1
- exact_match_naive
- prefix_match
- suffix_match
- equal_match

So missing, glue code, lua code, testing, so hopefully done till the end of next week

**Comment by mherna** [comment]
@Conni2461 should I close this issue and just wait for you to finish? 😄

**Comment by Conni2461** [comment]
Keep it open (i am able to close it if i want :smile:). Maybe in the end it will not work out :laughing: 

Also i would prefer if this will be fixed (even though tami wants to deprecate it). We should also port these options:
```lua
minimum_grep_characters = 2,
minimum_files_characters = 2,
```

Btw i am making progress and i am able to parse prompt input:
![image](https://user-images.githubusercontent.com/15233006/114314323-6f048200-9afa-11eb-92c8-3f68f6711674.png)

**Comment by tami5** [comment]
>  Maybe in the end it will not work out

Oh well at least our horrible agreement will be dropped and I will be in peace the following weeks :laughing: :laughing: 

btw awesome work conni I really admire how much work you put into this so far.

**Comment by Conni2461** [comment]
A hacky proof of concept works. Still needs more work.
![image](https://user-images.githubusercontent.com/15233006/114317688-99117080-9b09-11eb-8ccb-939a6dea2ab3.png)
![image](https://user-images.githubusercontent.com/15233006/114317731-ca8a3c00-9b09-11eb-9d3a-54a5d1369e5a.png)

**Comment by Conni2461** [comment]
Status update abou `telescope-fzf-native.nvim` progress for everyone subbed to this thread:
![image](https://user-images.githubusercontent.com/15233006/114577581-68057d00-9c7c-11eb-965c-2bfe4a4af175.png)

Missing:
- invert
- and probably more, i am not thinking about right now :rofl:

**Comment by yujinyuz** [comment]
> Status update abou `telescope-fzf-native.nvim` progress for everyone subbed to this thread:
> ![image](https://user-images.githubusercontent.com/15233006/114577581-68057d00-9c7c-11eb-965c-2bfe4a4af175.png)
> 
> Missing:
> 
> * invert
> * and probably more, i am not thinking about right now 🤣

thanks @Conni2461 hoping to get this soon! i miss these features when i was using fzf

**Comment by Conni2461** [comment]
ignore the last image that one was a LIE and is actually wrong thanks to @jesseleite for pointing that out.
![image](https://user-images.githubusercontent.com/15233006/114776023-4686bd00-9d72-11eb-89ab-2858c62a854d.png)
![image](https://user-images.githubusercontent.com/15233006/114776026-48e91700-9d72-11eb-90df-839e2709ec48.png)
![image](https://user-images.githubusercontent.com/15233006/114776058-54d4d900-9d72-11eb-9862-b3ef228e0ee6.png)
![image](https://user-images.githubusercontent.com/15233006/114776068-57373300-9d72-11eb-8b15-52488828f69e.png)

`smart_case`, `ignore_case`, `respect_case` is (hopefully) also implemented. `smart_case` example:
![image](https://user-images.githubusercontent.com/15233006/114776211-85b50e00-9d72-11eb-9997-2c7fb5aad549.png)
![image](https://user-images.githubusercontent.com/15233006/114776240-8baaef00-9d72-11eb-813f-f027429e7414.png)

Release hopefully in the next 24 hours. I will fix one more bug and maybe write some tests.

**Comment by jesseleite** [comment]
@Conni2461, You are my literal hero.

**Comment by Conni2461** [comment]
https://github.com/nvim-telescope/telescope-fzf-native.nvim

Read the friendly readme ;)

Also highlighting for `prefix_match` does not work because how telescope handles things. But i didn't want that you have to wait until eternity :laughing: i will fix that at some point

**Comment by Conni2461** [comment]
forgot to mention requires up to date telescope :)

**Comment by mherna** [comment]
@Conni2461 can you add to the readme an example on how to make a `live grep` using something like `require('telescope').extensions.fzf_native.live_grep()` :)

**Comment by Conni2461** [comment]
oh i think you are misunderstanding something. Its a sorter for everything besides `live_grep`. similar to `fzy_native`. `live_grep` doesn't have a sorter it sends each prompt (after each key press) to rg and displays these results.
`fzf_writer.live_grep` adds `fzf --filter` to this somehow, but i have never looked at that code. 

So this issue still needs to be fixed. Sorry if i was confusing about what i am working on :)

**Comment by mherna** [comment]
> oh i think you are misunderstanding something. Its a sorter for everything besides `live_grep`. similar to `fzy_native`. `live_grep` doesn't have a sorter it sends each prompt (after each key press) to rg and displays these results.
> `fzf_writer.live_grep` adds `fzf --filter` to this somehow, but i have never looked at that code.
> 
> So this issue still needs to be fixed. Sorry if i was confusing about what i am working on :)

Okay I understand, it's still super useful and way faster than `fzy_native`, thanks Conni!!!

**Comment by jesseleite** [comment]
>it's still super useful and way faster than fzy_native, thanks Conni!!!

Indeed, thanks Conni! PS. Is it actually _faster_ than the fzy_native extension?

**Comment by mherna** [comment]
Thanks @tjdevries for fixing it :)
